main.py

```diff
@@ -1,122 +1,151 @@
-                            "Error on line "
-                            + str(lineNum)
-                            + ": Insufficient number of coordinates. Please enter <lat long>"
-                        )
-                        break
-                    locations.append(coords)
-            f.close()
-
-    flashing = False
-    UDPOut.sendLED(mbedIP, mbedPort, "r")
-    rover.driveAlongCoordinates(locations, id1, id2)
-
-    if id1 != -1:
-        rover.trackARMarker(id1, id2)
-
-    flashing = True
-    lights = threading.Thread(target=flash)
-    lights.start()
-    # UDPOut.sendLED(mbedIP, mbedPort, 'g')
-
-    f = open("Recorded_Coordinates_" + args.latLong + ".txt", "a")
-    f.write("Latitude: " + rover.gps.latitude + "\n  Longitude: " + rover.gps.longitude)
-    f.close()
+# TODO: make this a part of a `Rover` class :3
+def drive(
+    rover: Drive, gps_coordinates: list[Tuple[float, float]], aruco_ids: list[int]
+) -> bool:
+    """
+    Given a Drive object, navigate to goal
 
+    return True if we make it the goal and False if we do not
+    """
 
-if __name__ == "__main__":
-    os.chdir(path)
-    print(os.getcwd())
-    # user input (args from system)
-    if args.cameraInput is None:
-        print("ERROR: must at least specify one camera")
-        exit(-1)
-
-    # gets the mbed ip and port
-    config = configparser.ConfigParser(allow_no_value=True)
-    if not config.read("config.ini"):
-        print("DID NOT OPEN CONFIG")
-        exit(-2)
-    mbedIP = str(config["CONFIG"]["MBED_IP"])
-    mbedPort = int(config["CONFIG"]["MBED_PORT"])
+    # Drive along GPS coordinates
+    rover.drive_along_coordinates(gps_coordinates, aruco_ids[0], aruco_ids[1])
+
+    # TODO: Can we have this run in the rover class instead of returning from 'drive_along_coordinates'
+    if aruco_ids[0] == -1:
+        rover.track_AR_Marker(aruco_ids[0], aruco_ids[1])
 
-    rover = Drive.Drive(50, args.cameraInput)
+    return True
+    # TODO: Return False? If we don't make it?
 
-    drive(rover)
+
+if __name__ == "__main__":
+    # Try to read from configuration file
+    config: configparser = configparser.ConfigParser(allow_no_value=True)
+    if not config.read("config.ini"):
+        raise Exception(
+            "Failed to open configuration file! Please ensure `config.ini` exists in this directory."
+        )
+
+    # Get MBED IP and MBED Port
+    mbed_ip: str = str(config["CONFIG"]["MBED_IP"])
+    mbed_port: int = int(config["CONFIG"]["MBED_PORT"])
+
+    # Read command line arguments
+    aruco_ids, gps_coordinates = parse_arguments()
+
+    # Initailze REMI!!
+    rover: Drive = Drive.Drive(50, args.cameraInput)
+
+    UDPOut.send_LED(
+        mbed_ip, mbed_port, "r"
+    )  # Change the LED to red to show REMI is entering Autonomous mode
+
+    # Drive to our goal and if we succeed, flash the LEDs green
+    if drive(rover, gps_coordinates, aruco_ids):
+        logger.info("Made it goal! Flash the LEDs green!")
```

Comment:
Extremely small thing, but missing the word "to" in "Made it to goal!"

Reply:
fixed in 770f335